flaskapp/tests.py

```diff
@@ -73,373 +54,191 @@ def test_get_deliveries_multiple(self):
-        self.create_dummy_targets()
-        self.client.delete(route)
+    def test_post_deliveries(self):
+        self.add_data_single()
+        r = self.post_data_single()
+        self.assertEquals(r.status_code, 200)
+        self.check_response_in_range(r)
 
-        data = [{
-            'description': 'Blood sample for patient Jane Doe',
-            'priority': 0,
-            'from': 1,
-            'to': 2
-        }]
-        r = self.client.post(route, data = json.dumps(data[0]))
+    def test_post_deliveries_no_description(self):
+        self.add_data_single()
+        r = self.post_data_single()
+        self.assertEquals(r.status_code, 200)
+        self.check_response_in_range(r)
 
+    def test_post_deliveries_error_no_name(self):
+        self.add_data_single()
+        del self.data[0]['name'] #Remove name value from data
+        r = self.post_data_single()
         self.assertEquals(r.status_code, 400)
 
     def test_post_deliveries_error_name_not_string(self):
-        route = '/deliveries'
-        self.create_dummy_targets()
-        self.client.delete(route)
-
-        data = [{
-            'name': None,
-            'description': 'Blood sample for patient',
-            'priority': 0,
-            'from': 1,
-            'to': 2
-        }]
-        r = self.client.post(route, data = json.dumps(data[0]))
-
+        self.add_data_single()
+        self.data[0]['name'] = 1 #Set name value to some non-string val
+        r = self.post_data_single()
         self.assertEquals(r.status_code, 400)
 
     def test_post_deliveries_error_no_priority(self):
-        route = '/deliveries'
-        self.create_dummy_targets()
-        self.client.delete(route)
-
-        data = [{
-            'name': 'Random name',
-            'description': 'Blood sample for patient',
-            'from': 1,
-            'to': 2
-        }]
-        r = self.client.post(route, data = json.dumps(data[0]))
-
+        self.add_data_single()
+        del self.data[0]['priority'] #Remove priority value from data
+        r = self.post_data_single()
         self.assertEquals(r.status_code, 400)
 
     def test_post_deliveries_error_invalid_priority(self):
-        route = '/deliveries'
-        self.create_dummy_targets()
-        self.client.delete(route)
-
-        data = [{
-            'name': 'Random name',
-            'description': 'Blood sample for patient',
-            'priority': None,
-            'from': 1,
-            'to': 2
-        }]
-        r = self.client.post(route, data = json.dumps(data[0]))
-
+        self.add_data_single()
+        self.data[0]['priority'] = None #Set priority value from data to None
+        r = self.post_data_single()
         self.assertEquals(r.status_code, 400)
 
     def test_delete_deliveries_empty(self):
-        route = '/deliveries'
-        self.create_dummy_targets()
-        self.client.delete(route)
-        r = self.client.delete(route)
+        r = self.client.delete(self.route)
```

Comment:
We might need to add a line here to check that the deliveries queue is actually empty.

`self.client.delete(self.route)`

IDEA: Run `self.client.delete(self.route)` before each test.